packages/components/src/assay/AssayGridPanel.tsx

```diff
@@ -0,0 +1,463 @@
+    if (showQCButton) manageItemCount++;
+    const showManageBtn = manageItemCount > 1;
+
+    if (!showImportBtn && manageItemCount == 0 && !showSampleBtn) {
+        return null;
+    }
+
+    return (
+        <div className="responsive-btn-group">
+            <div className="btn-group">
+                {showImportBtn && <AssayImportDataButton />}
+                {!showManageBtn && showDeleteBtn && (
+                    <DisableableButton
+                        bsStyle="default"
+                        onClick={onDelete}
+                        disabledMsg={
+                            !model.hasSelections ? 'Select one or more ' + nounPlural.toLowerCase() + '.' : undefined
+                        }
+                    >
+                        <span className="fa fa-trash" />
+                        <span>&nbsp;Delete</span>
+                    </DisableableButton>
+                )}
+                {!showManageBtn && showQCButton && (
+                    <UpdateQCStatesButton
+                        model={model}
+                        actions={actions}
+                        assayContainer={protocol.container}
+                        disabled={!model.hasSelections}
+                        requireCommentOnQCStateChange={assayDefinition.requireCommentOnQCStateChange}
+                    />
+                )}
+            </div>
+            {showManageBtn && (
+                <ManageDropdownButton id="assay-rows-grid" title="Edit" className="responsive-menu">
+                    {showQCButton && (
+                        <RequiresPermission perms={PermissionTypes.QCAnalyst}>
+                            <UpdateQCStatesButton
+                                model={model}
+                                actions={actions}
+                                assayContainer={protocol.container}
+                                disabled={!model.hasSelections}
+                                asMenuItem
+                                requireCommentOnQCStateChange={assayDefinition.requireCommentOnQCStateChange}
+                            />
+                            {(showEditBtn || showDeleteBtn) && <MenuItem divider />}
+                        </RequiresPermission>
+                    )}
+                    <RequiresPermission perms={PermissionTypes.Update}>
+                        <SelectionMenuItem
+                            id="assay-item-update-menu-item"
+                            text="Edit in Grid"
+                            onClick={toggleEditWithGridUpdate}
+                            maxSelection={MAX_EDITABLE_GRID_ROWS}
+                            queryModel={model}
+                            nounPlural={noun.toLowerCase()}
+                        />
+                        <SelectionMenuItem
+                            id="assay-item-bulk-update-menu-item"
+                            text="Edit in Bulk"
+                            onClick={showBulkUpdate}
```

Comment:
does this Edit in Bulk menu item need `maxSelection={ASSAY_MAX_EDITABLE_GRID_ROWS}` as well? just saw it was there in the LKB version of AssayResultsButtons.tsx

Reply:
I don't think so. You can still edit in bulk even if you can't edit in the grid.

Reply:
was it there because of the pathway to get to the "Edit in Grid" via the "Edit in Bulk" modal button?

Reply:
Possibly, but the check for enabling or disabling the button to go to the grid happens in the modal itself.